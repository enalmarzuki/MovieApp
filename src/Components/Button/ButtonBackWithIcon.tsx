import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import React from 'react';
import {Image, StyleSheet, TouchableOpacity} from 'react-native';
import {
  heightPercentageToDP,
  widthPercentageToDP,
} from 'react-native-responsive-screen';
import {ICArrowLeft} from '../../Assets/icons';
import {RootStackParamList} from '../../Router';

const ButtonBackWithIcon = () => {
  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();
  return (
    <TouchableOpacity
      style={styles.btnBack}
      onPress={() => navigation.goBack()}>
      <Image source={ICArrowLeft} testID="btn-back-icon" />
    </TouchableOpacity>
  );
};

export default ButtonBackWithIcon;

const styles = StyleSheet.create({
  btnBack: {
    backgroundColor: 'rgba(0,0,0,0.25)',
    padding: 2,
    borderRadius: 4,
    position: 'absolute',
    left: widthPercentageToDP('3%'),
    top: heightPercentageToDP('2%'),
    zIndex: 99,
  },
});
