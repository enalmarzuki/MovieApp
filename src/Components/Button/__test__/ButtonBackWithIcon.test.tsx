import {render} from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import ButtonBackWithIcon from '../ButtonBackWithIcon';
jest.useFakeTimers();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
    useRoute: () => ({}),
  };
});

describe('should render ButtonBackWithIcon', () => {
  it('should render correctly', () => {
    const BtnBackSnap = renderer.create(<ButtonBackWithIcon />).toJSON();
    expect(BtnBackSnap).toMatchSnapshot();
  });

  it('ButtonBackWithIcon must have icon arrow', () => {
    const {getByTestId} = render(<ButtonBackWithIcon />);
    expect(getByTestId('btn-back-icon')).toBeTruthy();
  });
});
