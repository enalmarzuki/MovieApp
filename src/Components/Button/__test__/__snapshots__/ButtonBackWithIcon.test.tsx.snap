// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ButtonBackWithIcon should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "rgba(0,0,0,0.25)",
      "borderRadius": 4,
      "left": 22.5,
      "opacity": 1,
      "padding": 2,
      "position": "absolute",
      "top": 26.5,
      "zIndex": 99,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/Assets/icons/ic-arrow-left.png",
      }
    }
    testID="btn-back-icon"
  />
</View>
`;
